src/main/java/io/jenkins/plugins/security/scan/bridge/BridgeInstall.java

```diff
@@ -16,13 +18,44 @@ public BridgeInstall(FilePath workspace, TaskListener listener) {
         this.logger = new LoggerWrapper(listener);
     }
 
-    public void installBridgeCLI(FilePath bridgeZipPath, FilePath bridgeInstallationPath)
+    public void installBridgeCLI(FilePath bridgeZipPath, FilePath bridgeInstallationPath, String subFolderName)
             throws PluginExceptionHandler {
         try {
             if (bridgeZipPath != null && bridgeInstallationPath != null) {
+                FilePath targetFolder = new FilePath(bridgeInstallationPath, subFolderName);
+
+                if (targetFolder.exists()) {
+                    // Delete the last folder from targetFolder
+                    logger.info("Deleting previous Bridge CLI folder: %s", targetFolder.getRemote());
+                    targetFolder.deleteRecursive();
+                }
+
                 logger.info("Unzipping Bridge CLI zip file from: %s", bridgeZipPath.getRemote());
                 bridgeZipPath.unzip(bridgeInstallationPath);
                 logger.info("Bridge CLI installed successfully in: %s", bridgeInstallationPath.getRemote());
+
+                // List all directories in bridgeInstallationPath matching the pattern
+                List<FilePath> matchingFolders = new ArrayList<>();
+                for (FilePath dir : bridgeInstallationPath.listDirectories()) {
```

Comment:
Since we are ensuring only one sub directory inside the parent folder we can form the sub directory name using the version provided by user. The logic should be:
- if version!="latest" then form the subfolder name like this `bridge-cli-bundle-VERSION-OSTYPE` 
- Check if `$HOME/bridge-cli-bundle/bridge-cli-bundle-VERSION-OSTYPE` exits
- if exists then rename `$HOME/bridge-cli-bundle/bridge-cli-bundle-VERSION-OSTYPE` to `$HOME/bridge-cli-bundle/bridge-cli-bundle-OSTYPE`

This will simplify this code. We are following the same for ADO and Action.